### PR TITLE
LL-5333 Market page provider

### PR DESCRIPTION
**Pull request by LFBarreto**
- LL-5333 (Market): add provider and api logic for market page

## Context (issues, jira)

[LL-5333]

## Expectations

- [ ] **Test coverage: The changes of this PR are covered by test.** Unit test were added with mocks when depending on a backend/device.
- [x] **No impact: The changes of this PR have ZERO impact on the userland.** Meaning, we can use these changes without modifying LLD/LLM at all. It will be a "noop" and the maintainers will be able to bump it without changing anything.

[LL-8218]: https://ledgerhq.atlassian.net/browse/LL-8218?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LL-5333]: https://ledgerhq.atlassian.net/browse/LL-5333?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/env.ts [hunk 1]
```diff
@@ -531,6 +531,11 @@ const envDefinitions = {
     parser: intParser,
     desc: "version used for the platform api",
   },
+  MARKET_API_URL: {
+    def: "https://api.coingecko.com/api/v3", // @TODO replace with prod API proxy whenever ready
```
Comment (gre) [inline src/env.ts:535]: that TODO makes this non mergeable, correct?

**Review by gre — CHANGES_REQUESTED** [review]
some CI issue

File: src/env.ts (hunk 1 above)
Comment (LFBarreto) [reply src/env.ts:535]: yes until we get actual infra proxy :grimacing:

**Review by MortalKastor — APPROVED** [review]

File: src/env.ts [hunk 2]
```diff
@@ -531,6 +531,11 @@ const envDefinitions = {
     parser: intParser,
     desc: "version used for the platform api",
   },
+  MARKET_API_URL: {
+    def: "https://proxycg.api.live.ledger.com/api/v3",
+    parser: stringParser,
+    desc: "url used to fetch platform catalog",
```
Comment (ofreyssinet-ledger) [inline src/env.ts:537]: is that the right description? 👀

**Review by OlivierFreyssinet — APPROVED** [review]

**Review by gre — APPROVED** [review]

**Pull request merged**